### ISSUE TITLE
Alerts for Google Cloud Monitoring

### ISSUE THREAD

**Issue by kylos101** [gitpod-io/gitpod#7859]
## Is your feature request related to a problem? Please describe

We lack the ability to alert via pager duty if a quota is becoming excessive for a project.

## Describe the behaviour you'd like

- [ ] Use a notification channel from Google to send a critical alert via pager duty if a quota is running low.
- [ ] A corresponding runbook should advise the on-caller to increase the quota, and if that doesn't resolve the issue in a timely manner, to create a case with Google Support, or escalate to Team Workspace if they are unsure.

## Describe alternatives you've considered

n/a

**Comment by ArthurSens** [comment]
This is currently blocked by https://github.com/gitpod-io/observability/issues/31 😕 

Just out of curiosity, what kind of metrics are you looking for that are available in Google Cloud Monitoring, and not in Prometheus?

**Comment by ArthurSens** [comment]
Hi @kylos101, this cannot be added on-demand unfortunately. 

To add a new datasource at the moment we need to expand our configuration surface at gitpod-io/observability. There is an issue requesting a way to configure Google Cloud Monitoring with grafana, which is the one mentioned above: gitpod-io/observability#31

It's an old issue that has been deprioritized a couple of times, but it's finally scheduled for next week 🙂

Can I ask you again, what do you expect to get there? 😅 I ask because prometheus has almost everything, only lacking logs and database metrics (interesting for webapp team), if its none of both maybe we already have what you want and I can help you get the information

**Comment by kylos101** [comment]
@ArthurSens hey Arthur, great question, I was thinking about it recently to help get you more context.

[Given these metrics](https://cloud.google.com/monitoring/api/metrics_gcp?hl=en_US), I think it would be good to alert if there is trouble with a quota (compute, storage, etc.), which would impact our ability to scale. Refer to metrics starting with `quota/*`.

@aledbf was there something else we cannot monitor now, that you were hoping to gain the ability to monitor with the support of GCE as a data source?

**Comment by mads-hartmann** [comment]
@kylos101 the work is being tracked in https://github.com/gitpod-io/observability/issues/31
@ArthurSens I'm assigning you to this, please leave a comment here and close it once the data source is available in production☺️

**Comment by ArthurSens** [comment]
I believe those issues are a little bit different mads. Here they want new dashboards and alerts, which is indeed something that will be done by the workspace team.

What we(platform) still need to do is make preview environments have stackdriver datasources so other teams can develop their dashboards.

Alerts with google cloud monitoring can be developed already, using terraform :)

**Comment by mads-hartmann** [comment]
@ArthurSens Thanks for clarifying! So this issue is (partially) blocked by https://github.com/gitpod-io/observability/issues/31. Once that issue is done the Stack Driver data source should be available for the teams to use as they like in their dashboards, is that right? ☺️

**Comment by ArthurSens** [comment]
exactly!

**Comment by mads-hartmann** [comment]
@ArthurSens Awesome, then I'll leave it assigned to you, and once the data source is available please unassign yourself and remove the issue form the Platform project ☺️

**Comment by ArthurSens** [comment]
The stackdriver datasource is working for previews and will land in [production](https://github.com/gitpod-io/ops/pull/950)/[staging](https://github.com/gitpod-io/ops/pull/949) soon :)

**Comment by kylos101** [comment]
> Alerts with google cloud monitoring can be developed already, using terraform :)

Cool! @ArthurSens can you share how we are doing that now with terraform? :pray:

**Comment by wulfthimm** [comment]
@kylos101 Please reach out to me when you have any question.

**Comment by mads-hartmann** [comment]
@kylos101 I'm removing this from the Platform project. Please re-add it if you want to follow up ☺️

**Comment by kylos101** [comment]
What @ArthurSens shared is perfect, thank you, @mads-hartmann and @wulfthimm .

**Comment by kylos101** [comment]
Hi @atduarte we were mostly concerned with getting alerts, if a quota is approaching an maximum, so that we can avoid service disruptions. This issue is dependent on [this one](https://github.com/gitpod-io/ops/issues/239#issuecomment-1092168860).

**Comment by sagor999** [comment]
@kylos101 I think this can be closed as done now? Since we do receive alerts for low quota now (not sure if they are pagerduty alerts though, but for sure saw them in slack alerts)

**Comment by kylos101** [comment]
@sagor999 agreed, if something more granular is needed, we can revisit in a new issue.
